kroxylicious/src/main/java/io/kroxylicious/proxy/future/BrutalFuture.java

```diff
@@ -0,0 +1,31 @@
+package io.kroxylicious.proxy.future;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class BrutalFuture<T> extends CompletableFuture<T> {
```

Comment:
I don't really like the name much.  This should not be part of the public API, so I would use a name like InternalFuture.

Reply:
sounds good, I'll rename to Internal. Maybe `FailsafeFuture`? It's hard to capture the intent for both non-blocking and the completion so maybe `Internal` is best.

Reply:
I think we should call it `Promise` or `NonBlockingFuture`, at a push `KroxyliciousPromise` (makes me think of a pinky promise 🃏) or `KroxyliciousFuture`. 

Personally `Internal*` in the public API makes no sense, if its internal its not part of the API

Reply:
It's internal in that it should be internal to `kroxylicious`, in the Filter API it is exposed as a `CompletionStage`, and will throw exceptions if Filter Authors do something like `stage.toCompletableFuture().get()` or `stage.toCompletableFuture().complete(x)`. The user's could potentially depend on the kroxylicious internals to get at the class.

One thing I'm a little worried about is surprising people. They might unit test against a `CompletableFuture` and everything works fine, but then when they install the Filter into kroxylicious they get a different behaviour. Maybe a comment on the method in the Filter API is enough.

Reply:
> One thing I'm a little worried about is surprising people. They might unit test against a CompletableFuture and everything works fine, but then when they install the Filter into kroxylicious they get a different behaviour.  Maybe a comment on the method in the Filter API is enough

+1 Solving the problem with docs.